src/audio/pcm_converter/pcm_converter.c

```diff
@@ -30,8 +30,11 @@ void pcm_convert_as_linear(const struct audio_stream *source, uint32_t ioffset,
 	int chunk;
 	int N1, N2;
 
-	assert(audio_stream_get_avail_samples(source) >= samples + ioffset);
-	assert(audio_stream_get_free_samples(sink) >= samples + ooffset);
+	/* assert enough avail/free samples in source and sink buffer */
+	if (audio_stream_get_avail_samples(source) < samples + ioffset)
+		return -EINVAL;
```

Comment:
Where do we trace the sink overrun or source underrun ?

Reply:
>Where do we trace the sink overrun or source underrun ?

In pcm converter context, client needs to check return code and trace possible error. This is already done in this PR, commit **dai: host: dma: Check returned value in dma_buffer_copy_* clients**